identity/identity.go

```diff
@@ -103,66 +64,115 @@ func GetIdentityHeader(ctx context.Context) string {
 	return ""
 }
 
-func checkHeader(id *XRHID, w http.ResponseWriter) error {
+var (
+	ErrMissingIdentity      = errors.New("missing x-rh-identity header")
+	ErrDecodeIdentity       = errors.New("unable to b64 decode x-rh-identity header")
+	ErrInvalidOrgIdIdentity = errors.New("x-rh-identity header has an invalid or missing org_id")
+	ErrMissingIdentityType  = errors.New("x-rh-identity header is missing type")
+)
 
+// CheckBasePolicy performs semantic identity check and returns nil for valid values or
+// an error instead.
+func CheckBasePolicy(id *XRHID) error {
```

Comment:
This is an exported function. Is the intention that a user will know what to use
it for? It's too is only called once within `DecodeIdentity`; does it need to
exist as a standalone function?

Reply:
I extrapolated the approach, I have nothing in my hands but I feel like it might be useful. But if the end result is a cleaner API then I agree, un-exported it.